xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/skin/SkinClassLoaderResource.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xpn.xwiki.internal.skin;
+
+import java.io.InputStream;
+
+import javax.inject.Provider;
+
+import org.xwiki.skin.ResourceRepository;
+import org.xwiki.url.URLConfiguration;
+
+import com.xpn.xwiki.XWikiContext;
+
+/**
+ * Skin resource for the {@link ClassLoaderSkin}, where the skin's resources are resolved using the class loader.
+ *
+ * @version $Id$
+ * @since 13.8RC1
+ */
+public class SkinClassLoaderResource extends AbstractSkinResource
```

Comment:
Hmm since this is the ClassLoader version of a SkinResource should the name be "ClassLoaderSkinResource" ?